src/misc/rocm_smi_wrap.cc

```diff
@@ -23,7 +23,7 @@ THE SOFTWARE.
 #include "core.h"
 #include "utils.h"
 
-static int is_wsl2 = -1;
+__thread static int is_wsl2 = -1;
```

Comment:
should it be `static __thread int is_wsl2 = -1;`? as per [https://gcc.gnu.org/onlinedocs/gcc/Thread-Local.html](https://gcc.gnu.org/onlinedocs/gcc/Thread-Local.html)
or the order does not matter?

Reply:
I am not language lawyer. But I think you may be right. I will change the order

Reply:
the other question is whether to use __thread vs. thread_local, the C++ construct for that.

Reply:
all other variables are defined using __thread in rccl/nccl. so better keep the consistency, I think

Reply:
then I will have to change my code :-)

Reply:
actually, it doesn't matter for this pr, but for mine it requires `thread_local`, since it is on a `std::vector` variable. I cannot compile it if I change to __thread, it gives me an error that it has a 'non-trivial destructor'

Reply:
that should be fine as well